code/modules/mob/living/carbon/xenomorph/castes/behemoth/abilities_behemoth.dm

```diff
@@ -1366,6 +1366,8 @@
 	if(istype(hit_object, /obj/structure/reagent_dispensers/fueltank))
 		var/obj/structure/reagent_dispensers/fueltank/hit_tank = hit_object
 		hit_tank.explode()
+	if(istype(hit_object, /obj/hitbox)) // These don't hit the vehicles, but rather their hitboxes, so isarmored will not work here
+		return on_hit_anything(get_turf(hit_object), proj)
 	return rock_broke(get_turf(hit_object), proj)
```

Comment:
I don't have a problem with what you're changing, but this whole behavior is just some weird ass stuff.
It does absolutely nothing if it hits an object (except specifically fuel tanks that just explode instantly?). Even if you make it hit tanks, it still excludes anything else like mechs at the very least? 
I assume not exploding when hitting other stuff is for walance, but if we want to hit tanks we should hit mechs as well.

Reply:
That's weird, cause the only difference between the rock throw rocks and landslide rocks is landslide calls on_hit_anything always, while throwing them has this exception. And it seems that when a rock collides with the tank, it just teleports to the tanks middle to hit it, thus always hitting it's hitbox

Reply:
Or at the very least, it always hits the tank no matter what part it hits